app/src/main/java/net/vonforst/evmap/api/nobil/NobilModel.kt

```diff
@@ -0,0 +1,234 @@
+    val apiversion: String = "3",
+    // val existingids: String,
+)
+
+@JsonClass(generateAdapter = true)
+data class NobilDetailSearchRequest(
+    val apikey: String,
+    val id: String,
+    val action: String = "search",
+    val type: String = "id",
+    val format: String = "json",
+    val apiversion: String = "3",
+)
+
+@JsonClass(generateAdapter = true)
+data class NobilResponseData(
+    @Json(name = "error") val error: String?,
+    @Json(name = "Provider") val provider: String?,
+    @Json(name = "Rights") val rights: String?,
+    @Json(name = "apiver") val apiver: String?,
+    @Json(name = "chargerstations") val chargerStations: List<NobilChargerStation>?
+)
+
+@JsonClass(generateAdapter = true)
+data class NobilChargerStation(
+    @Json(name = "csmd") val chargerStationData: NobilChargerStationData,
+    @Json(name = "attr") val chargerStationAttributes: NobilChargerStationAttributes
+) {
+    fun convert() = ChargeLocation(
+        chargerStationData.id,
+        "nobil",
+        HtmlCompat.fromHtml(chargerStationData.name, HtmlCompat.FROM_HTML_MODE_COMPACT).toString(),
+        chargerStationData.position,
+        Address(chargerStationData.city,
+                when (chargerStationData.landCode) {
+                    "DAN" -> "Denmark"
+                    "FIN" -> "Finland"
+                    "ISL" -> "Iceland"
+                    "NOR" -> "Norway"
+                    "SWE" -> "Sweden"
+                    else -> ""
+                },
+                chargerStationData.zipCode,
+                listOfNotNull(chargerStationData.street, chargerStationData.houseNumber).joinToString(" ")),
+        chargerStationAttributes.conn.map { createChargepointFromNobilConnection(it.value) },
+        null,
+        "",
+        null,
+        null,
+        chargerStationData.ocpiId != null ||
+                chargerStationData.updated.isAfter(LocalDateTime.now().minusMonths(6)),
+        null,
+        if (chargerStationData.operator != null) HtmlCompat.fromHtml(chargerStationData.operator, HtmlCompat.FROM_HTML_MODE_COMPACT).toString() else null,
+        if (chargerStationData.userComment != null) HtmlCompat.fromHtml(chargerStationData.userComment, HtmlCompat.FROM_HTML_MODE_COMPACT).toString() else null,
+        null,
+        if (chargerStationData.description != null) HtmlCompat.fromHtml(chargerStationData.description, HtmlCompat.FROM_HTML_MODE_COMPACT).toString() else null,
+        if (chargerStationData.image == "no.image.svg") null else listOf(NobilChargerPhotoAdapter(chargerStationData.image)),
+        null,
+        // 24: Open 24h
+        if (chargerStationAttributes.st["24"]?.attrTrans == "Yes") OpeningHours(twentyfourSeven = true, null, null) else null,
+        // 7: Parking fee
+        if (chargerStationAttributes.st["7"]?.attrTrans == "No") Cost(freeparking = true) else null,
+        null,
+        null,
+        null,
+        null,
+        Instant.now(),
+        true
+    )
```

Comment:
See https://github.com/ev-map/EVMap/pull/363/commits/70953af3b2aefe4cf0ee14c8f7f9cdfab59f267e. What do you think?

I kept the name "availability" although it could perhaps be confusing when the realtime data code refers to "availability" also.

Reply:
Looks good! Regarding the naming... Hm - maybe something like `access` instead of `availability`? i.e., a charger could be "available" (not in use) but not publicly "accessible".